### PR TITLE
Add APB role_type to ansible-galaxy init

### PR DESCRIPTION
**Pull request by dymurray**
##### SUMMARY

Adds `apb` role_type to `ansible-galaxy init`. This will allow Ansible-Galaxy users to initialize an APB project using the Ansible Galaxy CLI tool.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

Ansible Galaxy
##### ANSIBLE VERSION

```
ansible 2.6.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/dymurray/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible-2.6.0-py2.7.egg/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

```

##### ADDITIONAL INFORMATION

# Before
```
[dymurray@dymurray ansible]$ ansible-galaxy init -h
Usage: ansible-galaxy init [options] role_name

Options:
  --container-enabled   Initialize the skeleton role with default contents for
                        a Container Enabled role.
  -f, --force           Force overwriting an existing role
  -h, --help            show this help message and exit
  -c, --ignore-certs    Ignore SSL certificate validation errors.
  --init-path=INIT_PATH
                        The path in which the skeleton role will be created.
                        The default is the current working directory.
  --offline             Don't query the galaxy API when creating roles
  --role-skeleton=ROLE_SKELETON
                        The path to a role skeleton that the new role should
                        be based upon.
  -s API_SERVER, --server=API_SERVER
                        The API server destination
  -v, --verbose         verbose mode (-vvv for more, -vvvv to enable
                        connection debugging)
  --version             show program's version number and exit

 See 'ansible-galaxy <command> --help' for more information on a specific command.
```
# After
```
[dymurray@dymurray foo-apb]$ ansible-galaxy init -h
Usage: ansible-galaxy init [options] role_name

Options:
  -f, --force           Force overwriting an existing role
  -h, --help            show this help message and exit
  -c, --ignore-certs    Ignore SSL certificate validation errors.
  --init-path=INIT_PATH
                        The path in which the skeleton role will be created.
                        The default is the current working directory.
  --offline             Don't query the galaxy API when creating roles
  --role-skeleton=ROLE_SKELETON
                        The path to a role skeleton that the new role should
                        be based upon.
  -s API_SERVER, --server=API_SERVER
                        The API server destination
  --type=ROLE_TYPE      Type of skeleton role to initiatalize. Valid options:
                        'container', 'apb'.
  -v, --verbose         verbose mode (-vvv for more, -vvvv to enable
                        connection debugging)
  --version             show program's version number and exit

 See 'ansible-galaxy <command> --help' for more information on a specific command.
```

```
[dymurray@dymurray ansible-galaxy]$ ansible-galaxy init --type apb foo-apb
- foo-apb was created successfully
[dymurray@dymurray ansible-galaxy]$ ls foo-apb/
apb.yml  defaults  Dockerfile  files  Makefile  meta  playbooks  tasks  tests  vars
[dymurray@dymurray ansible-galaxy]$ 
```

### CODE REVIEW

**Comment by dymurray** [comment]
Instead of doing `ansible-galaxy init --container-enabled` a user would do `ansible-galaxy init --type container`

File: lib/ansible/cli/galaxy.py [hunk 1]
```diff
@@ -81,8 +81,8 @@ def set_action(self):
             self.parser.set_usage("usage: %prog init [options] role_name")
             self.parser.add_option('--init-path', dest='init_path', default="./",
                                    help='The path in which the skeleton role will be created. The default is the current working directory.')
-            self.parser.add_option('--container-enabled', dest='container_enabled', action='store_true', default=False,
-                                   help='Initialize the skeleton role with default contents for a Container Enabled role.')
+            self.parser.add_option('--type', dest='role_type', action='store', default='default',
+                                   help="Type of skeleton role to initiatalize. Valid options: 'container', 'apb'.")
```
Comment (Akasurde) [inline lib/ansible/cli/galaxy.py]: s/initiatalize/initialize/

File: lib/ansible/galaxy/data/apb/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Requirements
+------------
+
+Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance, if the role uses the EC2 module, it may be a good idea to mention in this section that the boto package is required.
+
+APB Variables
+--------------
+
+A description of the settable variables for this APB should go here, including any variables that are in defaults/main.yml, vars/main.yml, apb.yml, and any variables that can/should be set via parameters to the role. Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
```
Comment (Akasurde) [inline lib/ansible/galaxy/data/apb/README.md]: s/ie./i.e.,/

**Comment by chouseknecht** [comment]
shipit

**Review by chouseknecht — APPROVED** [review]

**Pull request merged**